pkg/otlp/collector.go

```diff
@@ -3,6 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2021-present Datadog, Inc.
 
+// +build !serverless
```

Comment:
Do we really need this build tag? The otlp package is only used at `cmd/agent/app/run.go` and in `pkg/trace/config/apply.go` so, if the serverless agent doesn't use either of those, I would expect this not to be necessary

Reply:
I think we need this because this file requires the exposed method `FromAgentConfig`. I tried to add it as a no-op in the new no-op file but this function has the following signature `func FromAgentConfig(cfg config.Config) (PipelineConfig, error)` so it would require to redefine PipelineConfig as well I think

Reply:
Sorry, I did not mean in this specific file, I meant in all files on every file under `pkg/otlp`

Reply:
I think what @mx-psi wants to say is that since you've removed the OTLP pkg dep in the `cmd/agent/common/common.go` file, the OTLP pkg should not be included anymore in the Serverless Agent build already, so you probably don't need to tag these `pkg/otlp` files with the `!serverless` flag. But we may have missed something, I did not tried to build the serverless agent locally for my review.

Reply:
Per https://github.com/DataDog/datadog-agent/pull/10068#discussion_r761086560 it looks like the serverless agent binary includes the trace agent in some way, so that's why this is needed

Reply:
Got it, thanks for the precision.

Reply:
Exact! Thanks